tests/legacy/test_resource_warnings.py

```diff
@@ -17,10 +19,21 @@ def test_task_double_close(self, init_kwargs):
             with nidaqmx.Task(**init_kwargs) as task:
                 task.close()
 
+    @pytest.mark.library_only
     def test_unclosed_task(self, init_kwargs):
         """Test to validate unclosed tasks."""
         task = nidaqmx.Task(**init_kwargs)
         # Since __del__ is not guaranteed to be called, for the purposes of
         # consistent test results call __del__ manually.
         with pytest.warns(DaqResourceWarning):
             task.__del__()
+
+    @pytest.mark.grpc_only
+    def test_explicit_detach_grpc_task(self, init_kwargs):
+        """Test to validate grpc leak tasks."""
+        task = nidaqmx.Task(**init_kwargs)
+
+        with warnings.catch_warnings(record=True) as warnings_raised:
+            task.__del__()
+            if warnings_raised:
+                pytest.fail()
```

Comment:
Split the assertion into a separate section (arrange/act/assert sections are separated by a blank line) and use the `assert` statement. The `warnings_raised` variable retains its value after exiting the `with` block.

```
        task = nidaqmx.Task(**init_kwargs)

        with warnings.catch_warnings(record=True) as warnings_raised:
            task.__del__()

        assert len(warnings_raised) == 0
```

Reply:
Updated.